### DISCUSSION TITLE
Heterogeneous graph classification task

### DISCUSSION THREAD

**Discussion by symd** [pyg-team/pytorch_geometric#6167] [Q&A]
I have around 4000 heterodata objects in my dataset. But some nodes can be part of multiple graphs and all types of edges may not be present in all graphs. I have given a unique id for each node in each heterodata object. Do the node ids  for these nodes needs to be unique across dataset? 

While training  I get the following error:
 File "<eval_with_key>.5", line 110, in forward
    pool_3 = torch.add(pool__nodetype1, pool__nodetype2);  pool__nodetype1 = pool__nodetype2 = None
RuntimeError: The size of tensor a (64) must match the size of tensor b (62) at non-singleton dimension 0

I would like to know the reason and how to solve it

**Comment by rusty1s** [comment]
If there exists some graphs who miss node types or edge types, you need to create dummy tensors for them, e.g.:
```python
data[node_type].x = torch.empty(0, 64)  # A feature matrix with 64 features but zero nodes
data[edge_type].edge_index = torch.empty(2, 0, dtype=torch.long)  # A matrix with zero edges

**Reply by symd** [reply]
Thank you. I have tried the same. But still the issue is there. I have given the following code
data[node_type].node_id=torch.empty(0).long()
data[node_type].x = torch.empty(0, 64)  
data[edge_type].edge_index = torch.empty(2, 0, dtype=torch.long) 
I have noticed that  a number is repeating in ptr of one node type.  Is there any problem in giving node_id like that?
I tried giving node_id as a tensor([0])

**Reply by rusty1s** [reply]
Do you have a minimal example to reproduce?

**Reply by symd** [reply]
[edges.csv](https://github.com/pyg-team/pytorch_geometric/files/10184354/edges.csv)
[graph.csv](https://github.com/pyg-team/pytorch_geometric/files/10184358/graph.csv)
[nodes.csv](https://github.com/pyg-team/pytorch_geometric/files/10184362/nodes.csv)

**Reply by rusty1s** [reply]
How do I need to load this data in?

**Reply by symd** [reply]
it worked when I replace the line
data[node_type].node_id=torch.empty(0).long()

with 
data[node_type].node_id=torch.tensor([0])

Thank you
